chord_metadata_service/chord/ingest/phenopackets.py

```diff
@@ -1,23 +1,37 @@
 from __future__ import annotations
 
 import uuid
+from humps import decamelize
 
 from dateutil.parser import isoparse
 from decimal import Decimal
 from chord_metadata_service.chord.models import Project, ProjectJsonSchema, Dataset
 from chord_metadata_service.phenopackets import models as pm
-from chord_metadata_service.phenopackets.schemas import PHENOPACKET_SCHEMA
+from chord_metadata_service.phenopackets.schemas import PHENOPACKET_SCHEMA, VRS_REF_REGISTRY
 from chord_metadata_service.patients.values import KaryotypicSex
 from chord_metadata_service.restapi.schema_utils import patch_project_schemas
 from chord_metadata_service.restapi.types import ExtensionSchemaDict
-from chord_metadata_service.restapi.utils import iso_duration_to_years
+from chord_metadata_service.restapi.utils import time_element_to_years
 
 from .exceptions import IngestError
 from .resources import ingest_resource
 from .schema import schema_validation
 from .utils import map_if_list, query_and_check_nulls
+from typing import Any, Dict, Iterable, Optional, Union, Callable, TypeVar
+from django.db.models import Model
 
-from typing import Any, Dict, Iterable, Optional, Union
+# Generic TypeVar for django db models
+T = TypeVar('T', bound=Model)
+
+
+def _get_or_create_opt(key: str, data: dict, create_func: Callable[..., T]) -> Optional[T]:
```

Comment:
can this function be tested?

Reply:
also, in my branch for new workflows, python 3.8 is being deprecated - once this is merged, we should move away from `Optional`

Reply:
Added test data to cover this function. 
Nice for 3.8 deprecation, can I merge your branch in mine already to get rid of Optionals?

Reply:
it would be good to merge, although for new authz/workflows you need some other branches as well - i posted on slack about it yesterday, but if you have time to look it would be nice to consolidate some of it yeah